ambry-rest/src/main/java/com.github.ambry.rest/NettyRequest.java

```diff
@@ -52,9 +53,14 @@
  * A wrapper over {@link HttpRequest} and all the {@link HttpContent} associated with the request.
  */
 class NettyRequest implements RestRequest {
+  // If the write of at least {@code bufferWatermark} amount of data is unacknowledged, reading from the channel will be
+  // temporarily suspended. It will be resumed when the amount of data unacknowledged drops below this number. If this
+  // is <=0, it is assumed that there is no limit on the size of unacknowledged data.
+  static int bufferWatermark = -1;
```

Comment:
This variable is used in this class for all the watermark checking (`NettyRequest` does not have instance of `NettyConfig`). The variable itself is set in `NettyServer` based on the config. I thought this would the easiest thing to do since this value is constant across all instances of `NettyRequest` and it didn't make sense to pass it through the constructor.

Could you explain why you think this may not be required?